File: internal/sast/fixtures/java/Jwt_positive.java
import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;

public class JwtPositive {
    private static final String JWT_SECRET = "supersecretpassword12345"; // SC-JAVA-JWT-003

    public Object decodeUnsigned(String token) {
        return Jwts.parser().parseClaimsJwt(token); // SC-JAVA-JWT-001
    }

    public Object decodeAllowsNone(String token) {
        return Jwts.parser().setSigningKey(JWT_SECRET).parse(token, SignatureAlgorithm.NONE); // SC-JAVA-JWT-002
    }
}

File: internal/sast/fixtures/csharp/Jwt_positive.cs
using System.IdentityModel.Tokens.Jwt;
using Microsoft.IdentityModel.Tokens;

public class JwtPositive
{
    private static readonly string JwtSecret = "supersecretpassword12345"; // SC-CSHARP-JWT-003

    public JwtSecurityToken ReadUnsigned(string token)
    {
        var handler = new JwtSecurityTokenHandler();
        return handler.ReadJwtToken(token); // SC-CSHARP-JWT-001
    }

    public void DisableValidation(string token)
    {
        var handler = new JwtSecurityTokenHandler();
        var p = new TokenValidationParameters
        {
            ValidateSignature = false,
            ValidateLifetime = false,
        };
        handler.ValidateToken(token, p, out _); // SC-CSHARP-JWT-002
    }
}
